core/miner/storage_fsm/impl/precommit_batcher_impl.cpp

```diff
@@ -0,0 +1,146 @@
+      : max_delay_(max_time),
+        api_(std::move(api)),
+        miner_address_(miner_address),
+        closest_cutoff_(closest_cutoff){};
+
+  PreCommitBatcherImpl::PreCommitEntry::PreCommitEntry(
+      TokenAmount number, api::SectorPreCommitInfo info)
+      : deposit(std::move(number)), precommit_info(std::move(info)){};
+
+  outcome::result<std::shared_ptr<PreCommitBatcherImpl>>
+  PreCommitBatcherImpl::makeBatcher(
+      const Ticks &max_wait,
+      std::shared_ptr<FullNodeApi> api,
+      std::shared_ptr<libp2p::protocol::Scheduler> scheduler,
+      const Address &miner_address) {
+    std::shared_ptr<PreCommitBatcherImpl> batcher =
+        std::make_shared<PreCommitBatcherImpl>(
+            max_wait, std::move(api), miner_address, max_wait);
+
+    batcher->cutoff_start_ = std::chrono::system_clock::now();
+    batcher->logger_ = common::createLogger("batcher");
+    batcher->logger_->info("Bather have been started");
+    batcher->handle_ = scheduler->schedule(max_wait, [=]() {
+      batcher->sendBatch();
+      batcher->handle_.reschedule(max_wait);  // TODO: maybe in gsl::finally
+    });
+    return batcher;
+  }
+
+  void PreCommitBatcherImpl::sendBatch() {
+    std::unique_lock<std::mutex> locker(mutex_, std::defer_lock);
+    // TODO(Elestrias): [FIL-398] goodFunds = mutualDeposit + MaxFee; -  for
+    // checking payable
+
+    if (batch_storage_.size() != 0) {
+      logger_->info("Sending procedure started");
+      auto head = api_->ChainHead().value();
+      auto minfo = api_->StateMinerInfo(miner_address_, head->key).value();
+
+      PreCommitBatch::Params params = {};
+      for (const auto &data : batch_storage_) {
+        mutual_deposit_ += data.second.deposit;
+        params.sectors.push_back(data.second.precommit_info);
+      }
+      auto encodedParams = codec::cbor::encode(params);
+      if (encodedParams.has_error()) {
+        logger_->error("Error has occurred during parameters encoding: {}",
+                       encodedParams.error().message());
+      }
+      logger_->info("Sending data to the network...");
+      auto maybe_signed = api_->MpoolPushMessage(
+          vm::message::UnsignedMessage(
+              miner_address_,
+              minfo.worker,  // TODO: handle worker
+              0,
+              mutual_deposit_,
+              {},
+              {},
+              25,  // TODO (m.tagirov) Miner actor v5 PreCommitBatch
+              MethodParams{encodedParams.value()}),
+          kPushNoSpec);
+
+      if (maybe_signed.has_error()) {
+        logger_->error("Error has occurred during batch sending: {}",
+                       maybe_signed.error().message());  // TODO: maybe logs
+      }
+
+      mutual_deposit_ = 0;
+      batch_storage_.clear();
+    }
+    cutoff_start_ = std::chrono::system_clock::now();
+    logger_->info("Sending procedure completed");
+  }
+
+  outcome::result<void> PreCommitBatcherImpl::forceSend() {
```

Comment:
`outcome::success` is always a result, so 
```suggestion
  void PreCommitBatcherImpl::forceSend() {
```